File: rust/src/models/existing_product_ref.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::product_id::ProductId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ExistingProductRef {
    pub id: ProductId,
}

impl ExistingProductRef {
    pub fn new(id: ProductId) -> Self {
        Self { id }
    }
}

File: typescript/src/models/existingProductRef.ts
// this file is @generated
import { type ProductId, ProductIdSerializer } from "./productId";

export interface ExistingProductRef {
  id: ProductId;
}

export const ExistingProductRefSerializer = {
  _fromJsonObject(object: any): ExistingProductRef {
    return {
      id: ProductIdSerializer._fromJsonObject(object["id"]),
    };
  },

  _toJsonObject(self: ExistingProductRef): any {
    return {
      id: ProductIdSerializer._toJsonObject(self.id),
    };
  },
};
